nodebuilder/p2p/bitswap.go

```diff
@@ -32,36 +29,26 @@ const (
 func dataExchange(params bitSwapParams) exchange.Interface {
 	prefix := protocolID(params.Net)
 	net := network.NewFromIpfsHost(params.Host, &routinghelpers.Null{}, network.Prefix(prefix))
-	srvr := server.New(
-		params.Ctx,
-		net,
-		params.Bs,
-		server.ProvideEnabled(false), // we don't provide blocks over DHT
+
+	opts := []bitswap.Option{
+		// Server options
+		bitswap.ProvideEnabled(false), // we don't provide blocks over DHT
 		// NOTE: These below are required for our protocol to work reliably.
 		// // See https://github.com/celestiaorg/celestia-node/issues/732
-		server.SetSendDontHaves(false),
-	)
+		bitswap.SetSendDontHaves(false),
 
-	clnt := client.New(
-		params.Ctx,
-		net,
-		params.Bs,
-		client.WithBlockReceivedNotifier(srvr),
```

Comment:
so bc they're coupled again now, we don't need to specify this?

Reply:
yes. `client.WithBlockReceivedNotifier` is enabled inside bitswap.New constructor.